bql/grammar/grammar_test.go

```diff
@@ -145,6 +145,18 @@ func TestAcceptByParse(t *testing.T) {
 		 from ?b where {?s "old_predicate_1"@[,] ?o1.
 				?s "old_predicate_2"@[,] ?o2.
 				?s "old_predicate_3"@[,] ?o3};`,
+		// Test Deconstruct clause.
+		`deconstruct {?s "new_predicate"@[] ?o} in ?a from ?b where {?s "old_predicate"@[,] ?o} having ?s = ?o;`,
+		`deconstruct {?s "new_predicate"@[] ?o} in ?a from ?b where {?s "old_predicate"@[,] ?o};`,
+		`deconstruct {?s ?p ?o.
+			      _:v "_subject"@[] ?s.
```

Comment:
I found this a bit misleading. The main reason in my head is the following. When you query in the where clause, you query blank nodes if you need to. This means, that after all, you get them like any other node/regular binding. Hence, for the deconstruct part you would actually use the regular binding you have obtained from the where  clause query. 

The blank node notation, needs to be consistent. If you have a blank node placeholder it must generate a new unique blank node. That would mostly mean, generating non preexisting nodes by definition, hence, never deleting anything. 

For the reasons above, I would suggest remove the _:v notation from the syntax of destruct.